asdf_astropy/converters/coordinates/tests/helpers.py

```diff
@@ -0,0 +1,42 @@
+from astropy.tests.helper import assert_quantity_allclose
+import astropy.units as u
+import numpy as np
+
+
+def assert_representation_equal(a, b):
+    __tracebackhide__ = True
+
+    assert type(a) is type(b)
+    assert a.components == b.components
+    for component in a.components:
+        assert u.allclose(getattr(a, component), getattr(b, component))
+
+
+def assert_sky_coord_equal(a, b):
+    __tracebackhide__ = True
+
+    assert a.is_equivalent_frame(b)
+    assert a.representation_type is b.representation_type
+    assert a.shape == b.shape
+
+    assert_representation_equal(a.data, b.data)
```

Comment:
Not your issue, but tracking down where the data attribute is actually set is a major task, or I'm really slow. It looks like major league indirection going on there.

Reply:
There's a gnarly `SkyCoord.__getattr__` method:

https://github.com/astropy/astropy/blob/main/astropy/coordinates/sky_coordinate.py#L819-L820

I think it's delegating `data` to the frame?

Reply:
That's what it looked like to me when digging, but I was too annoyed to keep following it.